dotCMS/src/main/java/com/dotcms/rest/BundlePublisherResource.java

```diff
@@ -45,140 +52,178 @@
+			final String bundleFolder       = fileName.substring(0, fileName.indexOf(".tar.gz"));
+			final String sendingEndPoint = sendingEndPointByAddress != null ? sendingEndPointByAddress.getId() : remoteIP;
+			final PublishAuditStatus status = PublishAuditAPI.getInstance().updateAuditTable(
+					sendingEndPoint, sendingEndPoint, bundleFolder, true);
+
+			if(bundleName.trim().length() > 0) {
+				// save bundle if it doesn't exists
+				bundle = APILocator.getBundleAPI().getBundleById(bundleFolder);
+				if (bundle == null || bundle.getId() == null) {
+
+					bundle = new Bundle();
+					bundle.setId(bundleFolder);
+					bundle.setName(bundleName);
+					bundle.setPublishDate(Calendar.getInstance().getTime());
+					bundle.setOwner(APILocator.getUserAPI().getSystemUser().getUserId());
+					bundle.setForcePush(forcePush);
+					APILocator.getBundleAPI().saveBundle(bundle);
 				}
-    		}
-    	} catch (IOException e) {
-    		Logger.error(PublisherQueueJob.class,e.getMessage(),e);
-    	}
+			}
+
+			//Write file on FS
+			FileUtil.writeToFile(bundleStream, bundlePath + fileName);
+
+			//Start thread
 
-		return Response.status(HttpStatus.SC_INTERNAL_SERVER_ERROR).build();
+			if(!status.getStatus().equals(Status.PUBLISHING_BUNDLE)) {
+
+				// todo: shouldn't be handle by the DotConcurrent?
+				new Thread(new PublishThread(fileName, groupId, endpointId, status)).start();
+			}
+
+			return bundle;
+		} catch (Exception e) {
+
+			Logger.error(PublisherQueueJob.class, "Error caused by remote call of: "+remoteIP);
+			Logger.error(PublisherQueueJob.class,e.getMessage(),e);
+			throw e;
+		}
 	}
 
-    /**
+	private String generatedBundleFileName() {
+		return String.format("bundle_%d.tar.gz", System.currentTimeMillis());
+	}
+
+	private boolean isAdmin(final User user) {
+
+    	return null != user && user.isBackendUser() && user.isAdmin();
+	}
+
+	/**
      * Validates a received token
      *
      * @param token    Token to validate
      * @param remoteIP Sender IP
-     * @param mySelf   Current end point
+     * @param publishingEndPoint   Current end point
      * @return True if valid
      * @throws IOException If fails reading the security token
      */
-    public static boolean isValidToken ( String token, String remoteIP, PublishingEndPoint mySelf ) throws IOException {
+    public static boolean isValidToken (final String token,
+										final String remoteIP,
+										final PublishingEndPoint publishingEndPoint) throws IOException {
 
         //My key
-        Optional<String> myKey=PushPublisher.retriveEndpointKeyDigest(mySelf);
-        if(!myKey.isPresent()) {
-          return false;
-        }
-
-
-        return token.equals( myKey.get() );
+        final  Optional<String> endpointKeyDigest = PushPublisher.retriveEndpointKeyDigest(publishingEndPoint);
+        return endpointKeyDigest.isPresent()? token.equals( endpointKeyDigest.get() ): false;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ternary operators that can be simplified with || or &&](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5594625)